include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -506,22 +641,49 @@ __pattern_adjacent_find_ranges(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy
     return std::ranges::borrowed_iterator_t<_R>(std::ranges::begin(__r) + __idx);
 }
 
+template <typename _Comp, typename _Proj>
```

Comment:
It looks, we already defined a type with the same functionality above. And we can re-use one.